app/jobs/migrate_notifications_job.rb

```diff
@@ -0,0 +1,63 @@
+class MigrateNotificationsJob < ApplicationJob
+  queue_as :default
+
+  def perform(*args)
+    # Do something later
+  end
+end
+class MigrateNotificationsJob < ApplicationJob
+  queue_as :default
+
+  BATCH_SIZE = 10
+  LIMIT = 100
+
+  # Define the Notification model to access the old table
+  class Notification < ApplicationRecord
+    self.inheritance_column = nil
+  end
+  
+
+  def perform(*args)  
+    total_processed = 0
+
+    last_processed_at = get_last_processed_at
+
+    # Process notifications in batches  
+    Notification.order(created_at: :desc).where("created_at > ?", last_processed_at).limit(LIMIT).find_in_batches(batch_size: BATCH_SIZE) do |batch|
```

Comment:
```suggestion
Notification.where("created_at > ?", last_processed_at).limit(LIMIT).find_in_batches(batch_size: BATCH_SIZE) do |batch|
```

Reply:
@hvillero this part tripped both of us up eh!
- the issue is that we wanted to order by `(created_at: :desc)` **and then** have that list be done in batches. The thing is, `find_in_batches` sorts by `asc` on the Primary Key (i.e. the `id` column), meaning it would start at the `Notification` with `id: 1` and then go in ascending order.

Reply:
@pJeyakumar , I think we need  the order in the: 
```ruby
Notification.order(created_at: :desc).where("created_at > ?", ..
```
but let's test it.